File: Week06/python/maxSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        """
        最大子序列和 = 当前元素自身最大 + 或者 包含之前最大
        :param nums:
        :return:
        """
        #时间复杂对O(n)
        dp = nums
        for i in range(1, len(nums)):
            dp[i] = nums[i] + max(nums[i], 0)

        return max(dp)

File: Week06/go/MaxSubArray.go
package main

func maxSubArray(nums []int) int {
	max:=func( x,y int) int {
		if (x<=y) {
			return y;
		}
		return x;
	}
	ans:=nums[0]
	sum:=0
	for _,v := range(nums){
		if (sum<0) {
			sum=v
		}else {
			sum+=v
		}
		ans=max(ans,sum)

	}
	return ans;

}




